service/src/main/python/jcarbon/nvml/sampler.py

```diff
@@ -80,6 +86,22 @@ def create_report(samples):
             for first, second in zip(samples[signal_name], samples[signal_name][1:]):
                 interval = create_interval_with_timestamp(first, second)
 
+                for d in first['data']:
+                    data = SignalInterval.SignalData()
```

Comment:
I know wrote this poorly before, but you could consider rewriting this to use `data.CopyFrom(d)` to replace the loop over the metadata (https://protobuf.dev/getting-started/pythontutorial/#standard-message-methods).